src/ml_tooling/result/viz/classification_viz.py

```diff
@@ -33,7 +36,15 @@ def confusion_matrix(self, normalized: bool = True, **kwargs) -> plt.Axes:
 
         with plt.style.context(self._config.STYLE_SHEET):
             title = f"Confusion Matrix - {self._estimator_name}"
-            y_pred = self._estimator.predict(self._data.test_x)
+            if threshold is None:
```

Comment:
With default value, we can get rid of the conditional

Reply:
Are there any classifiers which don't implement `predict_proba`?

Reply:
Dont think so, im thinking more of multiclass classifiers, the shape of predict_proba depends on the number of classes (.classes_)

Reply:
Then it’s safe to always use `pred_proba`

Reply:
Was the result that not all classifiers have a predict_proba?

Reply:
Yes, not all classifiers have a predict_proba unfortunately, so i tried working around that